readme.md

```diff
@@ -1,48 +1,408 @@
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Getting Started
+
+```bash
 $ npm install node-pptx
 ```
 
-## Usage
+Let's create a very simple presentation with one slide.
 
 ```javascript
-const pptx = require('node-pptx');
+const PPTX = require('node-pptx');
+let pptx = new PPTX.Composer();
 
-// In the futures presentations will also be able to composed through the DSL
-new pptx.Presentation()
-  .compose(pres => {
+await pptx.compose(pres => {
+    pres.addSlide(slide => {
+        slide
+            .addText(text => {
+                text.value('Hello World');
+            });
+    });
+});
 
-    pres.title('My Presentation')
-    .addSlide(slide => {
+await pptx.save(`./hello-world.pptx`);
+```
+## General Conventions
+`node-pptx` has a friendly declarative DSL to quickly design a pptx file. This makes your JavaScript code very readable because, it allows you to visually segment and compartmentalize your code to the presentation element you are trying to edit. Here is a simple example of adding a text box to a slide:
 
-      slide
-      .addText(text => {
+```javascript
+await pres.addSlide(slide => {
 
+    // declarative way of adding an object
+    slide.addText(text => {
         text
-        .value('Hello world')
-        .x(10)
-        .y(0);
-      })
-      .addShape(shape => {
+            .value('Hello World!')
+            .x(100)
+            .y(50)
+            .fontFace('Alien Encounters')
+            .fontSize(20)
+            .textColor('CC0000')
+            .textWrap('none')
+            .textAlign('left')
+            .textVerticalAlign('center')
+            .line({ color: '0000FF', dashType: 'dash', width: 1.0 })
+            .margin(0);
+    });
+});
+```
+You can also achieve the same result using the more terse object-only format by supplying a configuration object instead of a function.
+
+*Note*: Not all presentation elements support the object-only format.
+
+```javascript
+slide.addText({ value: 'Link to google.com', x: 100, y: 50, href: 'http://www.google.com' });
+```
+
+## Presentations
+The following sections defines the various ways to read, compose, and write pptx files. `node-pptx` allows you to either create a brand new file, or modify an existing pptx file.
+
+### Creating a Presentation From Scratch
+```javascript
+const PPTX = require('node-pptx');
+let pptx = new PPTX.Composer();
+
+await pptx.compose(pres => {
+    pres.addSlide(slide => {
+        slide
+            .addText(text => {
+                text.value('Hello World');
+            });
+    });
+});
+
+await pptx.save(`./hello-world.pptx`);
+```
+
+### Modifying an existing Presentation
+If you would like to use or modify an existing pptx file, simply load it first.
+
+```javascript
+const PPTX = require('node-pptx');
+let pptx = new PPTX.Composer();
+
+await pptx.load(`./existing.pptx`);
+await pptx.compose(async pres => {
+    await pres.getSlide('slide1').addImage(image => {
```

Comment:
FYI - no need for await here since when adding images from files, the addImage function doesn't execute any async code. You need it for URL sources.